### PR TITLE
Private mode auth.

### PR DESCRIPTION
**Pull request by tydangelo18**

### CODE REVIEW

File: tests/auth.test.ts [hunk 1]
```diff
@@ -1,48 +1,76 @@
...
+    it('should return a cookie', async () => {
+      const res = await kwil.auth.authenticateKGW(kSigner);
+
+      expect(res.status).toBe(200);
+      expect(res.data?.cookie).toBeDefined();
+      expect(res.data?.result).toBeDefined();
+      cookie = objects.requireNonNil(res.data?.cookie);
+    });
+
+    // TODO => need a fix for cookie setting
```
Comment (KwilLuke) [inline tests/auth.test.ts]: Does this TODO need to be resolved before merging? @tydangelo18

Comment (tydangelo18) [reply tests/auth.test.ts]: this test is not needed @KwilLuke

File: src/utils/rlp.ts [hunk 2]
```diff
@@ -1,93 +1,276 @@
...
+    return encodeSingleArguments(action);
+  });
+}
+
+/**
+ *
+ * @param {ValueType[]} preparedAction - The values of the actions to be executed.
+ * @returns single EncodedValue (authenticatePrivateMode())
+ */
+export function encodeSingleArguments(preparedAction: ValueType[]): EncodedValue[] {
```
Comment (KwilLuke) [inline src/utils/rlp.ts]: The whole point of adding these two functions is that you don't need `encodeArguments` and `constructEncodedValues` anymore.

**Review by KwilLuke — APPROVED** [review]
LGTM!

**Pull request merged**